File: app/src/main/java/com/example/testapp/MainActivity.java
package com.example.testapp;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.EditText;

public class MainActivity extends Activity {

    Button btn;
    TextView output;
    EditText input;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btn = findViewById(R.id.btnClick);
        output = findViewById(R.id.outputText);
        input = findViewById(R.id.inputText);

        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                String text = input.getText().toString();

                if(text.isEmpty()){
                    output.setText("Type something first ⚠️");
                } else {
                    output.setText("You typed: " + text);
                }

            }
        });
    }
}

File: app/src/main/java/com/automatax1/MainActivity.kt
// Path: automatax1/app/src/main/java/com/automatax1/MainActivity.kt
package com.automatax1

import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    
    private lateinit var statusText: TextView
    private lateinit var btnOpenEditor: Button
    private lateinit var btnSettings: Button
    private lateinit var btnCheckPermission: Button
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        
        statusText = findViewById(R.id.statusText)
        btnOpenEditor = findViewById(R.id.btnOpenEditor)
        btnSettings = findViewById(R.id.btnSettings)
        btnCheckPermission = findViewById(R.id.btnCheckPermission)
        
        btnOpenEditor.setOnClickListener {
            startActivity(Intent(this, ScriptEditorActivity::class.java))
        }
        
        btnSettings.setOnClickListener {
            startActivity(Intent(this, SettingsActivity::class.java))
        }
        
        btnCheckPermission.setOnClickListener {
            checkAccessibilityPermission()
        }
        
        updateStatus()
    }
    
    private fun checkAccessibilityPermission() {
        val enabled = isAccessibilityServiceEnabled()
        if (!enabled) {
            Toast.makeText(this, "Please enable automatax1 in Accessibility Settings", Toast.LENGTH_LONG).show()
            startActivity(Intent(Settings.ACTION_ACCESSIBILITY_SETTINGS))
        } else {
            Toast.makeText(this, "✅ Accessibility is enabled!", Toast.LENGTH_SHORT).show()
        }
        updateStatus()
    }
    
    private fun isAccessibilityServiceEnabled(): Boolean {
        val enabledServices = Settings.Secure.getString(
            contentResolver,
            Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES
        ) ?: return false
        
        return enabledServices.contains(packageName)
    }
    
    private fun updateStatus() {
        val enabled = isAccessibilityServiceEnabled()
        statusText.text = if (enabled) {
            "✅ Accessibility: Enabled"
        } else {
            "❌ Accessibility: Disabled"
        }
    }
}